pkgs/applications/version-management/forgejo/default.nix

```diff
@@ -59,12 +65,60 @@ buildGoModule rec {
       --prefix PATH : ${lib.makeBinPath [ bash git gzip openssh ]}
   '';
 
+  passthru.updateScript = writeShellScript "update-forgejo" ''
+    set -o errexit
+    set -o pipefail
+
+    export PATH=${lib.makeBinPath [
+      common-updater-scripts
+      coreutils
+      curl
+      jq
+      nix
+    ]}
+
+    releases=$(mktemp --suffix ".json")
+    curl "https://codeberg.org/api/v1/repos/forgejo/forgejo/releases?draft=false&pre-release=false&limit=1" \
+      --silent \
+      --header "accept: application/json" \
+      --output "$releases"
+
+    stable=$(mktemp --suffix ".json")
+    jq '.[0]
+      | .tag_name[1:] as $version
+      | ("forgejo-src-\($version).tar.gz") as $filename
+      | { $version, html_url } + (.assets | map(select(.name | startswith($filename)) | {(.name | split(".") | last): .browser_download_url}) | add)' \
+      "$releases" >"$stable"
+
+    version=$(jq -r .version "$stable")
+    release_url=$(jq -r .html_url "$stable")
+
+    if [[ "${version}" = "$version" ]]; then
+      echo "No new version found (already at $version)"
+      exit 0
+    fi
+
+    echo "Release: $release_url"
+
+    checksum_file=$(mktemp)
+    checksum_url=$(jq -r .sha256 "$stable")
+    curl "$checksum_url" --silent --output "$checksum_file"
+
+    sha256=$(cut --delimiter " " --fields 1 "$checksum_file")
+    sri_hash=$(nix hash to-sri --type sha256 "$sha256")
+
+    archive_url=$(jq -r .gz "$stable")
+
+    update-source-version "${pname}" "$version" "$sri_hash" "$archive_url"
+  '';
```

Comment:
considering `writeShellApplication` does most of the setup for us, that seems like a better fit?:
```suggestion
  passthru.updateScript = let
    name = "update-forgejo";
  in  "${writeShellApplication {
    inherit name;
    runtimeInputs = [
      common-updater-scripts
      coreutils
      curl
      jq
      nix
    ];
    text = ''
      releases=$(mktemp --suffix ".json")
      curl "https://codeberg.org/api/v1/repos/forgejo/forgejo/releases?draft=false&pre-release=false&limit=1" \
        --silent \
        --header "accept: application/json" \
        --output "$releases"

      stable=$(mktemp --suffix ".json")
      jq '.[0]
        | .tag_name[1:] as $version
        | ("forgejo-src-\($version).tar.gz") as $filename
        | { $version, html_url } + (.assets | map(select(.name | startswith($filename)) | {(.name | split(".") | last): .browser_download_url}) | add)' \
        "$releases" >"$stable"

      version=$(jq -r .version "$stable")
      release_url=$(jq -r .html_url "$stable")

      if [[ "${version}" = "$version" ]]; then
        echo "No new version found (already at $version)"
        exit 0
      fi

      echo "Release: $release_url"

      checksum_file=$(mktemp)
      checksum_url=$(jq -r .sha256 "$stable")
      curl "$checksum_url" --silent --output "$checksum_file"

      sha256=$(cut --delimiter " " --fields 1 "$checksum_file")
      sri_hash=$(nix hash to-sri --type sha256 "$sha256")

      archive_url=$(jq -r .gz "$stable")

      update-source-version "${pname}" "$version" "$sri_hash" "$archive_url"
    '';
  }}/bin/${name}";
```

Reply:
Good call!
I just exchanged the `let name = ""; in` with `lib.getExe` :)